### ISSUE TITLE
support "Regularized Greedy forest" - can beat xgboost on some tasks

### ISSUE THREAD

**Issue by gugatr0n1c** [dmlc/xgboost#1341]
Hi all,

there is implementation of gradient boosting which can (in term of accuracy) can beat xgboost in some tasks.

research paper:
http://stat.rutgers.edu/home/tzhang/software/rgf/tpami14-rgf.pdf

project website:
http://stat.rutgers.edu/home/tzhang/software/rgf/

Trick is to greedy optimized learning_rates and support  some more regularization.

If I understand it correctly, it can be used with xgboost as well (maybe with even with booster DART too).

If you consider this useful, it will be nice to see this inside xgboost.
Appriciate your thoutgs about this.

guga

**Comment by pommedeterresautee** [comment]
Not an issue.
